### PR TITLE
Add the revision support to api products

### PR DESCRIPTION
**Pull request by 1akshitha**
Added the support to import and export api products by revision Number and deploy based on the specified deployment environments. Implementation is similar to the api export and import.

Related Issue: wso2/product-apim-tooling#601 and wso2/product-apim#9406

### CODE REVIEW

File: import-export-cli/cmd/exportAPIProduct.go [hunk 1]
```diff
@@ -103,10 +112,14 @@ func init() {
 	ExportCmd.AddCommand(ExportAPIProductCmd)
 	ExportAPIProductCmd.Flags().StringVarP(&exportAPIProductName, "name", "n", "",
 		"Name of the API Product to be exported")
+	ExportAPIProductCmd.Flags().StringVarP(&exportAPIProductRevisionNum, "rev", "", "",
+		"Revision number of the API product to be exported")
```
Comment (wasuradananjith) [inline import-export-cli/cmd/exportAPIProduct.go:116]:
```suggestion
		"Revision number of the API Product to be exported")
```

Comment (wasuradananjith) [reply import-export-cli/cmd/exportAPIProduct.go:116]: Shall we change like this to be consistent?

File: import-export-cli/cmd/exportAPIProduct.go [hunk 2]
```diff
@@ -103,10 +112,14 @@ func init() {
...
 	ExportAPIProductCmd.Flags().StringVarP(&exportAPIProductProvider, "provider", "r", "",
 		"Provider of the API Product")
 	ExportAPIProductCmd.Flags().StringVarP(&CmdExportEnvironment, "environment", "e",
 		"", "Environment to which the API Product should be exported")
+	ExportAPIProductCmd.Flags().BoolVarP(&exportAPIProductLatestRevision, "latest", "", false,
+		"Export the latest revision of the API")
```
Comment (wasuradananjith) [inline import-export-cli/cmd/exportAPIProduct.go:122]:
```suggestion
		"Export the latest revision of the API Product")
```

Comment (wasuradananjith) [reply import-export-cli/cmd/exportAPIProduct.go:122]: Shall we correct this?

File: import-export-cli/cmd/getAPIProductRevisions.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+var getRevisionsAPIProductName string
+var getRevisionsAPIProductProvider string
+var getAPIProductRevisionsCmdEnvironment string
+var getAPIProductRevisionsCmdFormat string
+var getAPIProductRevisionsCmdQuery string
+
+// GetAPIProductRevisionsCmd related info
+const GetAPIProductRevisionsCmdLiteral = "api-product-revisions"
+const GetAPIProductRevisionsCmdShortDesc = "Display a list of Revisions for the API"
```
Comment (wasuradananjith) [inline import-export-cli/cmd/getAPIProductRevisions.go:37]:
```suggestion
const GetAPIProductRevisionsCmdShortDesc = "Display a list of Revisions for the API Product"
```

Shall we correct this? Check other places as well

File: import-export-cli/cmd/getAPIProductRevisions.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+const GetAPIProductRevisionsCmdLongDesc = `Display a list of Revisions available for the API in the environment specified`
```
Comment (Chamindu36) [inline import-export-cli/cmd/getAPIProductRevisions.go:39]: Shall we change this?
```suggestion
const GetAPIProductRevisionsCmdLongDesc = `Display a list of Revisions available for the API Product in the environment specified`
```

File: import-export-cli/cmd/getAPIProductRevisions.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+var getRevisionsCmdExamples = utils.ProjectName + ` ` + GetCmdLiteral + ` ` + GetAPIProductRevisionsCmdLiteral + ` -n PizzaAPI -v 1.0.0 -e dev
```
Comment (Chamindu36) [inline import-export-cli/cmd/getAPIProductRevisions.go:41]: Shall we change the example as well?

**Review by wasuradananjith — APPROVED** [review]

**Review by Chamindu36 — APPROVED** [review]

**Review by uvindra — APPROVED** [review]

**Pull request merged**